pkg/yurthub/kubernetes/meta/restmapper.go

```diff
@@ -0,0 +1,259 @@
+				return currKind, nil
+			}
+		}
+	}
+	return schema.GroupVersionKind{}, fmt.Errorf("no matches for %v", resource)
+}
+
+// Used to delete the mapping relationship between GVR and GVK in dynamicRESTMapper
+func (rm *RESTMapperManager) deleteKind(gvk schema.GroupVersionKind) error {
+	kindName := strings.TrimSuffix(gvk.Kind, "List")
+	plural, singular := meta.UnsafeGuessKindToResource(gvk.GroupVersion().WithKind(kindName))
+	rm.Lock()
+	delete(rm.dynamicRESTMapper, plural)
+	delete(rm.dynamicRESTMapper, singular)
+	rm.Unlock()
+	return rm.updateCachedDynamicRESTMapper()
+}
+
+// Used to update local files saved on disk
+func (rm *RESTMapperManager) updateCachedDynamicRESTMapper() error {
+	if rm.storage == nil {
+		return nil
+	}
+	rm.RLock()
+	d, err := marshalDynamicRESTMapper(rm.dynamicRESTMapper)
+	rm.RUnlock()
+	if err != nil {
+		return err
+	}
+	return rm.storage.Update(CacheDynamicRESTMapperKey, d)
+}
+
+// KindFor is used to find GVK based on GVR information.
+// 1. return true means the GVR is a built-in resource in shceme.
+// 2.1 return false and non-empty GVK means the GVR is custom resource
+// 2.2 return false and empty GVK means the GVR is unknown resource.
+func (rm *RESTMapperManager) KindFor(gvr schema.GroupVersionResource) (bool, schema.GroupVersionKind) {
+	gvk, kindErr := rm.unsafeDefaultRESTMapper.KindFor(gvr)
+	if kindErr != nil {
+		gvk, kindErr = rm.dynamicKindFor(gvr)
+		if kindErr != nil {
+			return false, schema.GroupVersionKind{}
+		}
+		return false, gvk
+	}
+	return true, gvk
+}
+
+// DeleteKindFor is used to delete the GVK information related to the incoming gvr
+func (rm *RESTMapperManager) DeleteKindFor(gvr schema.GroupVersionResource) error {
+	isScheme, gvk := rm.KindFor(gvr)
+	if !isScheme && !gvk.Empty() {
+		return rm.deleteKind(gvk)
+	}
+	return nil
+}
+
+// UpdateKind is used to verify and add the GVK and GVR mapping relationships of new Custom Resource
+func (rm *RESTMapperManager) UpdateKind(gvk schema.GroupVersionKind) error {
+	kindName := strings.TrimSuffix(gvk.Kind, "List")
+	gvk = gvk.GroupVersion().WithKind(kindName)
+	plural, singular := meta.UnsafeGuessKindToResource(gvk.GroupVersion().WithKind(kindName))
+	// If it is not a built-in resource and it is not stored in DynamicRESTMapper, add it to DynamicRESTMapper
+	isScheme, t := rm.KindFor(singular)
+	if !isScheme && t.Empty() {
+		rm.Lock()
+		rm.dynamicRESTMapper[singular] = gvk
+		rm.dynamicRESTMapper[plural] = gvk
+		rm.Unlock()
+		return rm.updateCachedDynamicRESTMapper()
+	}
+	return nil
+}
+
+// ResetRESTMapper is used to clean up all cached GVR/GVK information in DynamicRESTMapper,
+// and delete the corresponding file in the disk (cache-crd-restmapper.conf), it should be used carefully.
+func (rm *RESTMapperManager) ResetRESTMapper() error {
+	rm.dynamicRESTMapper = make(map[schema.GroupVersionResource]schema.GroupVersionKind)
+	err := rm.storage.DeleteCollection("_internal/restmapper")
```

Comment:
`_internal/restmapper` is hard code, please define a constant var.

Reply:
The code is updated, please review.